src/changelist/_format.py

```diff
@@ -3,21 +3,71 @@
 from collections import OrderedDict
 from collections.abc import Iterable
 from dataclasses import dataclass
-from typing import Union
+from datetime import datetime
 
 from github.NamedUser import NamedUser
 from github.PullRequest import PullRequest
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, slots=True)
+class ChangeItem:
+    content: str
+    reference_name: str
+    reference_url: str
+    labels: tuple[str]
+    timestamp: datetime
```

Comment:
This new data class (`PullRequest` -> `ChangeItem`) uncouples `_format.py` and `_query.py` a bit so that `MdFormatter` doesn't have to know about PRs at all. 

I intend to do the same for `NamedUser` -> `UserItem`. This should make it way easier to test the formatter without mocking. Because the PyGitHub objects are kinda opaque and not easy to initialize.

Long-term, might also help if the query part is moved to GraphQL queries.